### ISSUE TITLE
SQLFluff doesn't respect leading commas (L019) in macros (produces L005 errors)

### ISSUE THREAD

**Issue by Ofir-Purple** [sqlfluff/sqlfluff#3783]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When a file contains a macro call with leading commas (see below), SQLFluff returns an error `Commas should not have whitespace directly before them.` (L005) even though the project is configured with `comma_style=leading`. 

This error is produced only for leading commas inside the macro, and not for leading commas that are not inside a macro.

### Expected Behaviour

When `comma_style=leading` is set, the following model:

```SQL
SELECT * FROM (
    {{ my_simple_macro() }}
) AS tbl1

UNION

SELECT
    11 AS col1
    , 12 AS col2
    , 13 AS col3
```

with the following macro:

```SQL
{%- macro my_simple_macro() -%}
    SELECT
        1 AS col1
        , 2 AS col2
        , 3 AS col3
{%- endmacro -%}
```

Should not produce `L005` errors

### Observed Behaviour

```
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
11:35:08  Unable to do partial parsing because config vars, config profile, or config target have changed
11:35:08  Unable to do partial parsing because env vars used in profiles.yml have changed
11:35:08  Unable to do partial parsing because a project config has changed
== [vim-dbt/models/my_simple_model.sql] FAIL
L:   2 | P:   5 | L005 | Commas should not have whitespace directly before them.
L:   2 | P:   5 | L005 | Commas should not have whitespace directly before them.
All Finished 📜 🎉!
```

### How to reproduce

1. Create a dbt project with the above model and macro files
2. Add the following configuration to `.sqlfluff`:
```
[sqlfluff]
templater = dbt
dialect = postgres
ignore_templated_areas = False

[sqlfluff:rules]
comma_style=leading
```
3. run `sqlfluff lint`

### Dialect

dbt + Postgres

### Version

sqlfluff, version 1.3.0
sqlfluff-templater-dbt==1.3.0
dbt-core==1.0.3
Python 3.8.12

### Configuration

```

[sqlfluff]
templater = dbt
dialect = postgres
ignore_templated_areas = False

[sqlfluff:rules]
comma_style=leading

```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Hi @Ofir-Purple - I think I know what might be going on here. Would you mind running `sqlfluff parse` on the file in question and pasting the results here? I think somehow the `newline` elements at the star of lines are being stripped (maybe?) so that it can't identify the whitespace before the commas as an indent - it just sees it at excess whitespace.

The parse tree would allow much better debugging of this issue.

**Comment by Ofir-Purple** [comment]
Hi @alanmcruickshank, thanks for the reply. Yes, of course:

```bash
% sqlfluff parse models/my_simple_model.sql
=== [dbt templater] Compiling dbt project...
10:25:16  [WARNING]: Deprecated functionality
The `source-paths` config has been renamed to `model-paths`. Please update your
`dbt_project.yml` configuration to reflect this change.
10:25:16  [WARNING]: Deprecated functionality
The `data-paths` config has been renamed to `seed-paths`. Please update your
`dbt_project.yml` configuration to reflect this change.
=== [dbt templater] Project Compiled.
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        set_expression:
[L:  1, P:  1]      |            select_statement:
[L:  1, P:  1]      |                select_clause:
[L:  1, P:  1]      |                    keyword:                                  'SELECT'
[L:  1, P:  7]      |                    [META] indent:
[L:  1, P:  7]      |                    whitespace:                               ' '
[L:  1, P:  8]      |                    select_clause_element:
[L:  1, P:  8]      |                        wildcard_expression:
[L:  1, P:  8]      |                            wildcard_identifier:
[L:  1, P:  8]      |                                star:                         '*'
[L:  1, P:  9]      |                whitespace:                                   ' '
[L:  1, P: 10]      |                [META] dedent:
[L:  1, P: 10]      |                from_clause:
[L:  1, P: 10]      |                    keyword:                                  'FROM'
[L:  1, P: 14]      |                    whitespace:                               ' '
[L:  1, P: 15]      |                    from_expression:
[L:  1, P: 15]      |                        [META] indent:
[L:  1, P: 15]      |                        from_expression_element:
[L:  1, P: 15]      |                            table_expression:
[L:  1, P: 15]      |                                bracketed:
[L:  1, P: 15]      |                                    start_bracket:            '('
[L:  1, P: 16]      |                                    [META] indent:
[L:  1, P: 16]      |                                    newline:                  '\n'
[L:  2, P:  1]      |                                    whitespace:               '    '
[L:  2, P:  5]      |                                    select_statement:
[L:  2, P:  5]      |                                        select_clause:
[L:  2, P:  5]      |                                            keyword:          'SELECT'
[L:  2, P: 28]      |                                            [META] indent:
[L:  2, P:  5]      |                                            newline:          '\n'
[L:  2, P:  5]      |                                            whitespace:       '        '
[L:  2, P:  5]      |                                            select_clause_element:
[L:  2, P:  5]      |                                                numeric_literal:  '1'
[L:  2, P:  5]      |                                                whitespace:   ' '
[L:  2, P:  5]      |                                                alias_expression:
[L:  2, P:  5]      |                                                    keyword:  'AS'
[L:  2, P:  5]      |                                                    whitespace:  ' '
[L:  2, P:  5]      |                                                    naked_identifier:  'col1'
[L:  2, P:  5]      |                                            newline:          '\n'
[L:  2, P:  5]      |                                            whitespace:       '        '
[L:  2, P:  5]      |                                            comma:            ','
[L:  2, P:  5]      |                                            whitespace:       ' '
[L:  2, P:  5]      |                                            select_clause_element:
[L:  2, P:  5]      |                                                numeric_literal:  '2'
[L:  2, P:  5]      |                                                whitespace:   ' '
[L:  2, P:  5]      |                                                alias_expression:
[L:  2, P:  5]      |                                                    keyword:  'AS'
[L:  2, P:  5]      |                                                    whitespace:  ' '
[L:  2, P:  5]      |                                                    naked_identifier:  'col2'
[L:  2, P:  5]      |                                            newline:          '\n'
[L:  2, P:  5]      |                                            whitespace:       '        '
[L:  2, P:  5]      |                                            comma:            ','
[L:  2, P:  5]      |                                            whitespace:       ' '
[L:  2, P:  5]      |                                            select_clause_element:
[L:  2, P:  5]      |                                                numeric_literal:  '3'
[L:  2, P:  5]      |                                                whitespace:   ' '
[L:  2, P:  5]      |                                                alias_expression:
[L:  2, P:  5]      |                                                    keyword:  'AS'
[L:  2, P:  5]      |                                                    whitespace:  ' '
[L:  2, P:  5]      |                                                    naked_identifier:  'col3'
[L:  2, P: 28]      |                                        [META] dedent:
[L:  2, P: 28]      |                                    newline:                  '\n'
[L:  3, P:  1]      |                                    [META] dedent:
[L:  3, P:  1]      |                                    end_bracket:              ')'
[L:  3, P:  2]      |                            whitespace:                       ' '
[L:  3, P:  3]      |                            alias_expression:
[L:  3, P:  3]      |                                keyword:                      'AS'
[L:  3, P:  5]      |                                whitespace:                   ' '
[L:  3, P:  6]      |                                naked_identifier:             'tbl1'
[L:  3, P: 10]      |                        [META] dedent:
[L:  3, P: 10]      |            newline:                                          '\n'
[L:  4, P:  1]      |            newline:                                          '\n'
[L:  5, P:  1]      |            set_operator:
[L:  5, P:  1]      |                keyword:                                      'UNION'
[L:  5, P:  6]      |            newline:                                          '\n'
[L:  6, P:  1]      |            newline:                                          '\n'
[L:  7, P:  1]      |            select_statement:
[L:  7, P:  1]      |                select_clause:
[L:  7, P:  1]      |                    keyword:                                  'SELECT'
[L:  7, P:  7]      |                    [META] indent:
[L:  7, P:  7]      |                    newline:                                  '\n'
[L:  8, P:  1]      |                    whitespace:                               '    '
[L:  8, P:  5]      |                    select_clause_element:
[L:  8, P:  5]      |                        numeric_literal:                      '11'
[L:  8, P:  7]      |                        whitespace:                           ' '
[L:  8, P:  8]      |                        alias_expression:
[L:  8, P:  8]      |                            keyword:                          'AS'
[L:  8, P: 10]      |                            whitespace:                       ' '
[L:  8, P: 11]      |                            naked_identifier:                 'col1'
[L:  8, P: 15]      |                    newline:                                  '\n'
[L:  9, P:  1]      |                    whitespace:                               '    '
[L:  9, P:  5]      |                    comma:                                    ','
[L:  9, P:  6]      |                    whitespace:                               ' '
[L:  9, P:  7]      |                    select_clause_element:
[L:  9, P:  7]      |                        numeric_literal:                      '12'
[L:  9, P:  9]      |                        whitespace:                           ' '
[L:  9, P: 10]      |                        alias_expression:
[L:  9, P: 10]      |                            keyword:                          'AS'
[L:  9, P: 12]      |                            whitespace:                       ' '
[L:  9, P: 13]      |                            naked_identifier:                 'col2'
[L:  9, P: 17]      |                    newline:                                  '\n'
[L: 10, P:  1]      |                    whitespace:                               '    '
[L: 10, P:  5]      |                    comma:                                    ','
[L: 10, P:  6]      |                    whitespace:                               ' '
[L: 10, P:  7]      |                    select_clause_element:
[L: 10, P:  7]      |                        numeric_literal:                      '13'
[L: 10, P:  9]      |                        whitespace:                           ' '
[L: 10, P: 10]      |                        alias_expression:
[L: 10, P: 10]      |                            keyword:                          'AS'
[L: 10, P: 12]      |                            whitespace:                       ' '
[L: 10, P: 13]      |                            naked_identifier:                 'col3'
[L: 10, P: 17]      |                [META] dedent:
[L: 10, P: 17]      |    newline:                                                  '\n'
[L: 11, P:  1]      |    [META] end_of_file:

%
```
Thanks again 🙏

**Comment by alanmcruickshank** [comment]
Yes - I think my theory is right. It's not detecting the newlines before the commas as newlines (because they're coming from the macro), so not excluding the whitespace as indents (as they would normally). I think this should be very solvable.

**Comment by alanmcruickshank** [comment]
I believe this issue is now resolved. Testing on the current `main`, this statement passes without raising an `LT04` (which is the successor to `L019`), and regardless of comma config we don't get any spacing issues (`LT01`, which is the successor to `L005`):

```sql
-- sqlfluff:layout:type:comma:line_position:leading

{% macro my_simple_macro() -%}
    SELECT
        1 AS col1
        , 2 AS col2
        , 3 AS col3
{%- endmacro -%}

SELECT * FROM (
    {{ my_simple_macro() }}
) AS tbl1

UNION

SELECT
    11 AS col1
    , 12 AS col2
    , 13 AS col3
```

I'm going to close this issue as resolved. Do feel free to open if you can recreate with a different query on one of the more recent versions.
